arctic/store/_version_store_utils.py

```
@@ -76,6 +80,219 @@ def _define_compat_pickle_load():
+       or dangling segments from a failed append.
+    Parameters
+    ----------
+    l : `arctic.store.version_store.VersionStore`
+        The VersionStore instance against which the analysis will be run.
+    sym : `str`
+        The symbol to test if is corrupted.
+    input_v : `int` or `arctic.store.version_store.VersionedItem`
+        The specific version we wish to test if is appendable. This argument is mandatory.
+
+    Returns
+    -------
+    `bool`
+        True if the symbol is safe to append, False otherwise.
+    """
+    input_v = l._versions.find_one({'symbol': sym, 'version': input_v}) if isinstance(input_v, int) else input_v
+    return not _fast_check_corruption(l._collection, sym, input_v, check_append_safe=True)
+
+
+def fast_is_corrupted(l, sym, input_v):
+    """
+    This method can be used for a fast check (not involving a read) for a corrupted version.
+    Users can't trust this as may give false negatives, but it this returns True, then symbol is certainly broken (no false positives)
+    Parameters
+    ----------
+    l : `arctic.store.version_store.VersionStore`
+        The VersionStore instance against which the analysis will be run.
+    sym : `str`
+        The symbol to test if is corrupted.
+    input_v : `int` or `arctic.store.version_store.VersionedItem`
+        The specific version we wish to test if is corrupted. This argument is mandatory.
+
+    Returns
+    -------
+    `bool`
+        True if the symbol is found corrupted, False otherwise.
+    """
+    input_v = l._versions.find_one({'symbol': sym, 'version': input_v}) if isinstance(input_v, int) else input_v
+    return _fast_check_corruption(l._collection, sym, input_v)
+
+
+def is_corrupted(l, sym, input_v):
+    """
+        This method can be used to check for a corrupted version.
+        Will resort to a regular read (slower) if the internally invoked fast-detection gives hint for a corruption.
```

Looks like it does the read if `_fast_check_corruption` returns False. i.e. it's slow in the non-corrupt case?

---

The API doc is misleading/wrong.

The _fast_check_corruption returns true if found corruption, and the method will fall back to full read only if the fast check suggests that there is no corruption:
```python
if not _fast_check_corruption(l._collection, sym, input_v):
```
i.e. we want to fail fast (w/o read) if we find corruption with the fast check, and perform the read only if the data look otherwise good form a count/max-segment perspective.

I will update the doc.